src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -193,16 +196,178 @@ public override SyntaxToken VisitToken(SyntaxToken token)
+                        // assumption proves incorrect.
+                        if (list[i].IsKind(SyntaxKind.WhitespaceTrivia))
+                        {
+                            // All preceeding whitespace, including elastic markers seen previously, are ignored
+                            continue;
+                        }
+                        else if (list[i].IsKind(SyntaxKind.EndOfLineTrivia))
+                        {
+                            blankLineCount++;
+                        }
+                        else
+                        {
+                            // A non-ignored trivia is encountered. Assume that the EndOfLine for this trivia is not
+                            // blank.
+                            return Math.Max(0, blankLineCount - 1);
+                        }
+                    }
+
+                    // The beginning of the list was reached, so no need to correct for a non-blank line.
+                    return blankLineCount;
+                }
+
+                static int getFollowingBlankLineCount(in SyntaxTriviaList list, int index)
+                {
+                    var blankLineCount = 0;
+                    for (var i = index + 1; i < list.Count; i++)
+                    {
+                        // The current line is assumed to be blank now that a node was removed, and is counted with the
+                        // following blank lines.
+                        if (list[i].IsKind(SyntaxKind.WhitespaceTrivia))
+                        {
+                            if (isElasticMarker(list[i]))
+                            {
+                                // This marker will be processed once it is reached
+                                break;
+                            }
+
+                            continue;
+                        }
+                        else if (list[i].IsKind(SyntaxKind.EndOfLineTrivia))
+                        {
+                            blankLineCount++;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+
+                    return blankLineCount;
+                }
+
+                static SyntaxTriviaList removeBlankLinesAfterMarker(SyntaxTriviaList list, int index, int linesToRemove)
+                {
+                    var linesRemoved = 0;
+                    for (int i = index + 1; linesRemoved < linesToRemove && i < list.Count; i++)
+                    {
+                        if (list[i].IsKind(SyntaxKind.EndOfLineTrivia))
+                        {
+                            list = list.RemoveAt(i);
+                            linesRemoved++;
+                            i--;
+                        }
+                    }
+
+                    return list;
+                }
+            }
```

Comment:
i didn't look deeply at this.  if we get good test coverage for lots of interesting cases, that works for me.